File: oj-ui/src/types/traceItem.ts
export type TraceStatus = 'RUNNING' | 'SUCCESS' | 'FAILED'

export type TraceItemType = 'LLM' | 'TOOL' | 'NODE'

export interface TraceItem {
  id: number
  traceId: string
  nodeName: string
  itemType: TraceItemType
  itemKey: string
  roundNo: number
  toolName: string | null
  inputPayload: string | null
  outputPayload: string | null
  inputSummary: string | null
  outputSummary: string | null
  promptTokens: number | null
  completionTokens: number | null
  totalTokens: number | null
  status: TraceStatus
  errorMessage: string | null
  startTimestamp: number | null
  endTimestamp: number | null
  createTime: string
  updateTime: string
}

export type TraceItemStatus = 'running' | 'success' | 'error'

// 将后端状态转换为前端展示状态。
export function getTraceItemStatus(item: Pick<TraceItem, 'status'>): TraceItemStatus {
  if (item.status === 'RUNNING') {
    return 'running'
  }
  if (item.status === 'SUCCESS') {
    return 'success'
  }
  return 'error'
}

// 将耗时数字格式化为便于阅读的文本。
export function formatTraceMetric(value: number | null | undefined): string {
  if (value === null || value === undefined) {
    return '-'
  }
  return `${value}`
}

File: agent-oj-core/src/main/java/com/oj/agent/core/trace/model/entity/TraceItem.java
package com.oj.agent.core.trace.model.entity;

import com.baomidou.mybatisplus.annotation.IdType;
import com.baomidou.mybatisplus.annotation.TableId;
import com.baomidou.mybatisplus.annotation.TableName;
import lombok.Data;

import java.time.LocalDateTime;

@Data
@TableName("trace_item")
public class TraceItem {

    @TableId(value = "id", type = IdType.AUTO)
    private Long id;

    private String traceId;

    private String nodeName;

    private String itemType;

    private String itemKey;

    private Integer roundNo;

    private String toolName;

    private String inputPayload;

    private String outputPayload;

    private String inputSummary;

    private String outputSummary;

    private Integer promptTokens;

    private Integer completionTokens;

    private Integer totalTokens;

    private String status;

    private String errorMessage;

    private Long startTimestamp;

    private Long endTimestamp;

    private LocalDateTime createTime;

    private LocalDateTime updateTime;
}
